pkg/util/http.go

```diff
@@ -203,11 +205,9 @@ func decompressRequest(buffers *RequestBuffers, reader io.Reader, expectedSize,
 	if expectedSize > maxSize {
 		return nil, MsgSizeTooLargeErr{Actual: expectedSize, Limit: maxSize}
 	}
-	if compression != NoCompression && compression != RawSnappy && compression != Gzip {
```

Comment:
I decided to refactor this to have all the logic about how to read the buffer on the same place.